File: sdk/examples/js/client_basic.ts
/**
 * Basic client — connect to agent, create task, stream output.
 *
 * Run: YAVER_URL=http://localhost:18080 YAVER_TOKEN=xxx npx tsx examples/js/client_basic.ts
 */
import { YaverClient } from "../../js/src/index";

const url = process.env.YAVER_URL || "http://localhost:18080";
const token = process.env.YAVER_TOKEN || "";
if (!token) {
  console.error("Set YAVER_TOKEN env var");
  process.exit(1);
}

const client = new YaverClient(url, token);

async function main() {
  // Health check
  console.log("Health:", await client.health());
  console.log(`Ping: ${await client.ping()}ms`);

  // Agent info
  const info = await client.info();
  console.log(`Connected to ${info.Hostname} (v${info.Version})\n`);

  // Create task
  const task = await client.createTask("List all TypeScript files in the current directory");
  console.log(`Task ${task.id} created (status: ${task.status})\n`);

  // Stream output
  for await (const chunk of client.streamOutput(task.id)) {
    process.stdout.write(chunk);
  }

  // Final result
  const final = await client.getTask(task.id);
  console.log(`\n\nTask finished (status: ${final.status})`);

  // Clean up
  await client.deleteTask(task.id);
}

main().catch(console.error);

File: sdk/examples/python/client_basic.py
"""Basic client — connect to agent, create task, stream output."""
import os
import sys
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "../../python"))
from yaver import YaverClient

url = os.environ.get("YAVER_URL", "http://localhost:18080")
token = os.environ.get("YAVER_TOKEN", "")
if not token:
    print("Set YAVER_TOKEN env var")
    sys.exit(1)

client = YaverClient(url, token)

# Health check
print("Health:", client.health())
print(f"Ping: {client.ping():.1f}ms")

# Create task
task = client.create_task("List all Python files in the current directory")
print(f"Task {task['id']} created (status: {task['status']})\n")

# Stream output
for chunk in client.stream_output(task["id"]):
    print(chunk, end="")

# Final result
final = client.get_task(task["id"])
print(f"\n\nTask finished (status: {final['status']})")
